src/main/java/uk/gov/hmcts/reform/slc/services/UpdateLetterStatusJob.java

```diff
@@ -15,20 +17,33 @@
 
     private final FtpClient ftpClient;
     private final FtpAvailabilityChecker ftpAvailabilityChecker;
+    private final SendLetterClient sendLetterClient;
+    private final ReportParser parser;
+
 
-    public UpdateLetterStatusJob(FtpClient ftpClient, FtpAvailabilityChecker ftpAvailabilityChecker) {
+    public UpdateLetterStatusJob(
+        FtpClient ftpClient,
+        FtpAvailabilityChecker ftpAvailabilityChecker,
+        SendLetterClient sendLetterClient,
+        ReportParser parser
+    ) {
         this.ftpClient = ftpClient;
         this.ftpAvailabilityChecker = ftpAvailabilityChecker;
+        this.sendLetterClient = sendLetterClient;
+        this.parser = parser;
     }
 
     @Scheduled(cron = "${reports-cron}")
     public void run() {
         logger.trace("Running job");
 
         if (ftpAvailabilityChecker.isFtpAvailable(now())) {
-            ftpClient.downloadReports();
-            // TODO: extract letter status
-            // TODO: send update request to letters service
+            ftpClient
+                .downloadReports()
+                .stream()
+                .map(parser::parse)
+                .flatMap(List::stream)
+                .forEach(sendLetterClient::updatePrintedAt);
```

Comment:
I'd be happier to see the lambda version of these last two lines, but up to you.